packages/amplify-graphql-api-construct/src/__tests__/__functional__/props.test.ts

```diff
@@ -69,52 +74,216 @@ describe('supports different props configurations', () => {
+          },
+        });
+      });
+    });
+
+    it('does not allow conflictResolution and dataStoreConfiguration', () => {
+      const stack = new cdk.Stack();
+      expect(() => {
+        new AmplifyGraphqlApi(stack, 'TestApi', {
+          definition: AmplifyGraphqlDefinition.fromString(/* GraphQL */ `
+            type Todo @model @auth(rules: [{ allow: public }]) {
+              description: String!
+            }
+          `),
+          authorizationModes: {
+            apiKeyConfig: { expires: cdk.Duration.days(7) },
+          },
+          conflictResolution: {
+            project: {
+              handlerType: 'AUTOMERGE',
+              detectionType: 'VERSION',
+            },
+          },
+          dataStoreConfiguration: {
+            project: {
+              handlerType: 'AUTOMERGE',
+              detectionType: 'VERSION',
+            },
+          },
+        });
+      }).toThrow(
+        'conflictResolution is deprecrated. conflictResolution and dataStoreConfiguration cannot be used together. Please use dataStoreConfiguration.',
+      );
+    });
+
+    it('conflictResolution and dataStoreConfiguration are equivalent', () => {
+      const executeTransformSpy = jest.spyOn(graphqlTransformer, 'executeTransform');
+      const dataStoreConfigurationTemplate = verifySynth((stack) => {
+        new AmplifyGraphqlApi(stack, 'TestApi', {
+          definition: AmplifyGraphqlDefinition.fromString(/* GraphQL */ `
+            type Todo @model @auth(rules: [{ allow: public }]) {
+              description: String!
+            }
+          `),
+          authorizationModes: {
+            apiKeyConfig: { expires: cdk.Duration.days(7) },
+          },
+          dataStoreConfiguration: {
+            project: {
+              handlerType: 'AUTOMERGE',
+              detectionType: 'VERSION',
+            },
+          },
+        });
+      });
+
+      const conflictResolutionTemplate = verifySynth((stack) => {
+        new AmplifyGraphqlApi(stack, 'TestApi', {
+          definition: AmplifyGraphqlDefinition.fromString(/* GraphQL */ `
+            type Todo @model @auth(rules: [{ allow: public }]) {
+              description: String!
+            }
+          `),
+          authorizationModes: {
+            apiKeyConfig: { expires: cdk.Duration.days(7) },
+          },
+          dataStoreConfiguration: {
+            project: {
+              handlerType: 'AUTOMERGE',
+              detectionType: 'VERSION',
+            },
+          },
+        });
+      });
+
+      expect(executeTransformSpy.mock.calls[0][0].resolverConfig).toEqual(executeTransformSpy.mock.calls[1][0].resolverConfig);
+      expect(dataStoreConfigurationTemplate.toJSON().Metadata).toEqual(conflictResolutionTemplate.toJSON().Metadata);
+    });
   });
 
-  it('supports conflict resolution with lambda handling', () => {
+  it('supports a definition with a SQLModelDataSourceBinding with a VPC configuration', () => {
     verifySynth((stack) => {
```

Comment:
Are line 286~303 related to the test description  in line 285?

Reply:
Oops, accidentally combined two tests.